source/features/tag-changelog-link.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+This feature adds a compare link on each releases/tags/single tag page so that you can see
+what has changed since the previous release. If the tags are namespaced then it tries to
+get the previous release of the same namespaced tag.
+
+See it in action at: https://github.com/parcel-bundler/parcel/releases
+*/
+
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import fetchDom from '../libs/fetch-dom';
+import {getOwnerAndRepo} from '../libs/utils';
+import {isSingleTagPage} from '../libs/page-detect';
+import {diff} from '../libs/icons';
+
+async function init(): Promise<void | false> {
+	if (select.exists('.blankslate')) {
+		return false;
+	}
+
+	// To extract the tag "v16.8.6" from links like "/facebook/react/releases/tag/v16.8.6"
+	const tagRegExp = /releases\/tag\/(.*)$/;
+
+	// To extract the commit hash from links like "/facebook/react/commit/92a1d8feac32d03ab5ea6ac13ae4941f6ae93b54"
+	const commitRegExp = /commit\/([0-9a-f]{40})$/;
+
+	const tagSelector = 'h4.commit-title a[href*="/releases/tag"], .commit .release-header a[href*="/releases/tag"]';
+	const commitIdSelector = '.commit > ul a[href*="/commit/"], .release > div:first-child > ul a[href*="/commit/"]';
+
+	const allTags = extractValuesFromPathname(select.all<HTMLAnchorElement>(tagSelector), tagRegExp);
+	const allCommitIdsAnchor = select.all<HTMLAnchorElement>(commitIdSelector);
+	const allCommitIds = extractValuesFromPathname(allCommitIdsAnchor, commitRegExp);
+
+	const nextPageLink = select<HTMLAnchorElement>('.pagination a:last-child');
+	const nextPage = nextPageLink ? await fetchDom(nextPageLink.href) : await getNextPageForSinglePageTag(allTags[0]);
+
+	if (nextPage) {
+		allTags.push(...extractValuesFromPathname(select.all<HTMLAnchorElement>(tagSelector, nextPage), tagRegExp));
+		allCommitIds.push(...extractValuesFromPathname(select.all<HTMLAnchorElement>(commitIdSelector, nextPage), commitRegExp));
+	}
+
+	const {ownerName, repoName} = getOwnerAndRepo();
+
+	for (let index = 0; index < allCommitIdsAnchor.length; index++) {
+		const prevTagIndex = getPreviousTagIndex(index, allCommitIds, allTags);
+
+		if (prevTagIndex !== -1) {
+			allCommitIdsAnchor[index].closest('ul')!.append(
+				<li className="d-inline-block mb-1 mt-1 f6">
+					<a
+						className="muted-link text-mono tooltipped tooltipped-n"
+						aria-label={'See changes since ' + allTags[prevTagIndex]}
+						href={`/${ownerName}/${repoName}/compare/${allTags[prevTagIndex]}...${allTags[index]}`}
+					>
+						{diff()} Changelog
+					</a>
+				</li>
+			);
+		}
+	}
+}
+
+const extractValuesFromPathname = (anchors: HTMLAnchorElement[], regexp: RegExp): string[] => {
+	return anchors.map((anchor: HTMLAnchorElement): string => {
+		const [, value] = anchor.pathname.match(regexp)!;
```

Comment:
```suggestion
		return anchor.textContent!.trim();
```

You might have to adjust the selector to select these links:

<img width="127" alt="" src="https://user-images.githubusercontent.com/1402241/57009786-5f504380-6c2b-11e9-8c9b-13e9518a53f2.png">